File: lab2/lab3.cpp
// Program to implement  BLE algorithm
#include<iostream>
#include<math.h>

using namespace std;

int main()
{
    float x1,y1,x2,y2,p0,xk,yk,pk,/*x_nxt,y_nxt,p_nxt,*/dy,dx,m;
    int i,steps;
    cout<<"Enter starting co-ordinates x1 and y1:";
    cin>>x1>>y1;
    cout<<"Enter ending co-ordonates x2 and y2:";
    cin>>x2>>y2;
    dx = x2 - x1;
    dy = y2 - y1;
    m = dy/dx;
   if(fabs(m) < 1)
   {
    steps = dx;
    p0 = 2*dy - dx;
    xk = x1;
    yk = y1;
    pk = p0;
    for(i = 0;i < steps;i++)
    {
        cout<<"Iteration:"<<i<<"\t"<<"Pk:"<<pk<<"\t"<<"("<<xk<<","<<yk<<")"<<endl;
        if(pk < 0)
        {
            
        }
    }
   }
}

File: lab2/lab3.py
import math
import matplotlib.pyplot as plt

x1 = float(input("Enter starting co-ordinate x1:"))
y1 = float(input("Enter starting co-ordinate y1:"))
x2 = float(input("Enter ending co-ordinate x2:"))
y2 = float(input("Enter ending co-ordinate y2:"))

plt.plot([x1,x2],[y1,y2])

dx = x2 - x1
dy = y2 - y1
m = dy/dx
xk = x1
yk = y1
plt.scatter(xk,yk)

if abs(m)<1:
    i = 1
    po = 2*dy - dx
    pk = po
    steps = dx
    while i<=steps:
        if pk < 0:
            xk+=1
            pk = pk + 2*dy
        else:
            xk+=1
            yk+=1
            pk = pk + 2*dy - 2*dx
        i=i+1
        print(f"xk:{xk}\tyk:{yk}")
        plt.scatter(xk,yk)
    plt.show()
    
else:
    i = 1
    po = 2 * dx - dy
    pk = po
    steps = dy
    while i<=steps:
        if pk < 0:
            yk+=1
            pk = pk + 2*dx
        else:
            xk+=1
            yk+=1
            pk = pk + 2*dx - 2*dy
        i=i+1
        print(f"xk:{xk}\tyk:{yk}")
        plt.scatter(xk,yk)
    plt.show()    

